docs/rules/no-unsafe-finally.md

```diff
@@ -0,0 +1,99 @@
+# Disable Unsafe Statements in `finally` Block (no-unsafe-finally)
+
+Using control flow operations in a `finally` block may lead to unexpected behavior. For example:
+
+```js
+try {
+   return 1;
+} catch(err) {
+   return 2;
+} finally {
+   return 3;
+}
+// > 3
+
+try {
+    throw new Error("Try");
+    return 1;
+} catch(err) {
+    throw err;
+    return 2;
+} finally {
+    throw new Error("Finally");
+    return 3;
+}
+
+// > Uncaught Error: Finally(...)
+```
+
+## Rule Details
+
+This rule forbids the usage of `return`, `throw`, `break` and `continue` operations inside a `finally` block, ignoring function declarations.
+
+Examples of **correct** code for this rule:
+
+```js
+/*eslint no-unsafe-finally: "error"*/
+let foo = function() {
+    try {
+        return 1;
+    } catch(err) {
+        return 2;
+    } finally {
+        console.log("hola!");
+    }
+};
+```
+
+```js
+/*eslint no-unsafe-finally: "error"*/
+let foo = function() {
+    try {
+        return 1;
+    } catch(err) {
+        return 2;
+    } finally {
+        let a = function() {
+            try {
+                return 1;
+            } catch(err) {
+                return 2;
+            } finally {
+                console.log("hola!");
```

Comment:
This is deep water for me, but from the “ignoring” qualification in the sentence under Rule Details, I expected to see within the function, not necessarily a nested `try` statement, but that an ordinary `return` statement is valid.